### ISSUE TITLE
Validation messages from annotations in interface

### ISSUE THREAD

**Issue by JChrist** [dropwizard/dropwizard#1932]
I wanted to put all JAX-RS and validation annotations in an interface and simply implement the interface in a resource class. This works fine, apart from the validation messages becoming:
`method_name.arg0 is not a valid email address`, whereas if the annotations were on the resource field, it would be: `form field username is not a valid email address`

**Comment by nickbabcock** [comment]
This like a very reasonable improvement to track, thank you for the submission.

**Comment by JChrist** [comment]
I'd be willing to give it a shot, if you could point me to the right direction for implementing this

**Comment by nickbabcock** [comment]
The [ConstraintMessage](https://github.com/dropwizard/dropwizard/blob/786e9dd54e0882bd74683345f663adfc82c6ce44/dropwizard-jersey/src/main/java/io/dropwizard/jersey/validation/ConstraintMessage.java) class would be where you want to start (and it's corresponding class [ConstraintViolationExceptionMapperTest](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-jersey/src/test/java/io/dropwizard/jersey/validation/ConstraintViolationExceptionMapperTest.java), though feel free to create additional classes as needed).

And if you're feeling really ambitious, we want some kind of mechanism to extend/customize validations to support use cases like #1782

**Comment by JChrist** [comment]
I haven't had much success yet, even the simplest test case fails with server error (500 response) when the annotations are on an interface. I tend to believe it is due to the grizzly test container, but I am not sure.

**Comment by nickbabcock** [comment]
I actually wasn't able to reproduce the issue. The validation annotations are being used. Here's the interface and resource

```java
@Path("/valid2/")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public interface RestInterface {
    @POST
    @Path("repr")
    @Valid
    ValidRepresentation repr(@NotNull @Valid ValidRepresentation representation,
                             @NotNull @QueryParam("interfaceVariable") String xer);
}

public class ValidatingResource2 implements RestInterface {
    @Override
    public ValidRepresentation repr(ValidRepresentation representation, String xer) {
        return representation;
    }
}
```

From my following test case diff

```diff
diff --git a/dropwizard-jersey/src/test/java/io/dropwizard/jersey/validation/ConstraintViolationExceptionMapperTest.java b/dropwizard-jersey/src/test/java/io/dropwizard/jersey/validation/ConstraintViolationExceptionMapperTest.java
index a6ee3e2..b3b11bd 100644
--- a/dropwizard-jersey/src/test/java/io/dropwizard/jersey/validation/ConstraintViolationExceptionMapperTest.java
+++ b/dropwizard-jersey/src/test/java/io/dropwizard/jersey/validation/ConstraintViolationExceptionMapperTest.java
@@ -6,6 +6,7 @@ import io.dropwizard.jersey.DropwizardResourceConfig;
 import io.dropwizard.jersey.jackson.JacksonMessageBodyProviderTest.Example;
 import io.dropwizard.jersey.jackson.JacksonMessageBodyProviderTest.ListExample;
 import io.dropwizard.jersey.jackson.JacksonMessageBodyProviderTest.PartialExample;
+import io.dropwizard.other.ValidatingResource2;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -27,13 +28,13 @@ import static org.junit.Assume.assumeThat;
 
 public class ConstraintViolationExceptionMapperTest extends AbstractJerseyTest {
 
-
     private static final Locale DEFAULT_LOCALE = Locale.getDefault();
 
     @Override
     protected Application configure() {
         return DropwizardResourceConfig.forTesting(new MetricRegistry())
                 .packages("io.dropwizard.jersey.validation")
+                .register(new ValidatingResource2())
                 .register(new HibernateValidationFeature(Validators.newValidator()));
     }
 
@@ -79,6 +80,15 @@ public class ConstraintViolationExceptionMapperTest extends AbstractJerseyTest {
     }
 
     @Test
+    public void postInvalidInterfaceEntityIs422() throws Exception {
+        final Response response = target("/valid2/repr").request(MediaType.APPLICATION_JSON)
+            .post(Entity.entity("{\"name\": \"a\"}", MediaType.APPLICATION_JSON));
+        assertThat(response.getStatus()).isEqualTo(400);
+        assertThat(response.readEntity(String.class))
+            .isEqualTo("{\"errors\":[\"query param interfaceVariable may not be null\"]}");
+    }
+
+    @Test
     public void returnInvalidEntityIs500() throws Exception {
         assumeThat(Locale.getDefault().getLanguage(), is("en"));
 
diff --git a/dropwizard-jersey/src/test/java/io/dropwizard/other/RestInterface.java b/dropwizard-jersey/src/test/java/io/dropwizard/other/RestInterface.java
new file mode 100644
index 0000000..0ed5992
--- /dev/null
+++ b/dropwizard-jersey/src/test/java/io/dropwizard/other/RestInterface.java
@@ -0,0 +1,23 @@
+package io.dropwizard.other;
+
+import io.dropwizard.jersey.validation.ValidRepresentation;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+@Path("/valid2/")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface RestInterface {
+    @POST
+    @Path("repr")
+    @Valid
+    ValidRepresentation repr(@NotNull @Valid ValidRepresentation representation,
+                             @NotNull @QueryParam("interfaceVariable") String xer);
+}
diff --git a/dropwizard-jersey/src/test/java/io/dropwizard/other/ValidatingResource2.java b/dropwizard-jersey/src/test/java/io/dropwizard/other/ValidatingResource2.java
new file mode 100644
index 0000000..fbb555d
--- /dev/null
+++ b/dropwizard-jersey/src/test/java/io/dropwizard/other/ValidatingResource2.java
@@ -0,0 +1,10 @@
+package io.dropwizard.other;
+
+import io.dropwizard.jersey.validation.ValidRepresentation;
+
+public class ValidatingResource2 implements RestInterface {
+    @Override
+    public ValidRepresentation repr(ValidRepresentation representation, String xer) {
+        return representation;
+    }
+}
```

**Comment by nickbabcock** [comment]
Actually this issue exists in 1.0.6, but will be fixed in 1.1.0. I'll add my previous snippet as a test case so we don't happen to regress at some point.
